eclass/kernel-install.eclass

```diff
@@ -525,7 +728,7 @@ kernel-install_pkg_prerm() {
 kernel-install_pkg_postrm() {
 	debug-print-function ${FUNCNAME} "${@}"
 
-	if [[ -z ${ROOT} ]] && use initramfs; then
+	if [[ -z ${ROOT} && ! ${KERNEL_IUSE_GENERIC_UKI} ]]; then
```

Comment:
please keep the original `use` check for old kernels though. 

That said, I'm surprised we don't need it. I don't think Portage is supposed to remove files if their contents have changed.

Reply:
To be honest the original use check doesn't make sense to me, if installkernel is used to generate some initrd without having USE=initramfs here, then we still want to remove it upon removing the kernel I think.

But this is a separate thing, so I'll restore it for now.

Reply:
Yeah, it dates back to the eclass generating initramfs. But I'd like to avoid any changes to the stable ebuilds for now.